library/modules/Gui.cpp

```diff
@@ -948,16 +1011,66 @@ df::unit *Gui::getAnyUnit(df::viewscreen *top)
 
         return vector_get(world->units.active, *ui_selected_unit);
     }
+    case ZonesPitInfo: // (i) zone -> (P)it
+    case ZonesPenInfo: // (i) zone -> pe(N)
+    {
+        if (!ui_building_assign_units || !ui_building_item_cursor)
+            return NULL;
+
+        return vector_get(*ui_building_assign_units, *ui_building_item_cursor);
+    }
+    case Burrows:
+    {
+        if (ui->burrows.in_add_units_mode)
+            return vector_get(ui->burrows.list_units, ui->burrows.unit_cursor_pos);
+
+        return NULL;
+    }
+    case QueryBuilding:
+    {
+        df::building *building = getAnyBuilding(top);
+        if (building->getType() == df::building_type::Cage)
+        {
+            auto *cage = static_cast<df::building_cagest*>(building);
+            if (*ui_building_item_cursor < cage->assigned_units.size())
```

Comment:
Could check for this global being non-null somewhere (not sure if I missed it earlier)